src/libraries/Native/Unix/CMakeLists.txt

```diff
@@ -152,15 +156,15 @@ endif(CLR_CMAKE_TARGET_FREEBSD)
 #       ./build-native.sh cmakeargs -DCLR_ADDITIONAL_COMPILER_OPTIONS=<...> cmakeargs -DCLR_ADDITIONAL_LINKER_FLAGS=<...>
 #
 if(CLR_CMAKE_TARGET_UNIX)
-    if (CLR_CMAKE_TARGET_DARWIN)
-        add_definitions(-DTARGET_DARWIN)
-        add_link_options(-Wl,-bind_at_load)
-    elseif (CLR_CMAKE_TARGET_ARCH_WASM)
-        # No object stripping for WASM
-    else (CLR_CMAKE_TARGET_DARWIN)
-        add_compile_options($<$<COMPILE_LANGUAGE:ASM>:-Wa,--noexecstack>)
-        add_link_options(-Wl,--build-id=sha1 -Wl,-z,relro,-z,now)
-    endif(CLR_CMAKE_TARGET_DARWIN)
+    if(NOT CLR_CMAKE_TARGET_ARCH_WASM AND NOT CLR_CMAKE_TARGET_IOS)
+        if(CLR_CMAKE_TARGET_OSX)
+            add_definitions(-DTARGET_OSX)
+            add_link_options(-Wl,-bind_at_load)
```

Comment:
@janvorli I see that you have added `-bind_at_load` in  https://github.com/dotnet/runtime/pull/685/files#diff-1e019a64c92fd71addd77bc8288ca11fR174 . Was it intentional? Why is it needed?

It looks like a unnecessary de-optimization to me.

Reply:
The `-bind_at_load` is an equivalent of `-z,now` that we use on Linux.

From https://gcc.gnu.org/onlinedocs/gcc-4.7.4/gcc/Darwin-Options.html:
> -bind_at_load
Causes the output file to be marked such that the dynamic linker will bind all undefined references when the file is loaded or launched.

From https://linux.die.net/man/1/ld
> now
When generating an executable or shared library, mark it to tell the dynamic linker to resolve all symbols when the program is started, or when the shared library is linked to using dlopen, instead of deferring function call resolution to the point when the function is first called.

Reply:
I understand what it does. I do not understand why we need it. What is not going to work when we remove `-z,now` and `-bind_at_load` ?

Reply:
It is a hardening feature. `-z,relro` enables just a partial relro. To enable full relro, both `-z,relro` and `-z,now` is needed. See https://www.redhat.com/en/blog/hardening-elf-binaries-using-relocation-read-only-relro. GOT ovewrite attacks are still possible with partial relro, but not with full relro.

Reply:
Maybe worth a comment.